addons/barcodes/static/src/components/manual_barcode.js

```diff
@@ -0,0 +1,42 @@
+import { BarcodeDialog } from '@web/core/barcode/barcode_dialog';
+import { Component, onMounted, useRef, useState } from "@odoo/owl";
+
+export class BarcodeInput extends Component {
+    static template = "barcodes.BarcodeInput";
+    static props = {
+        onSubmit: Function,
+        placeholder: { type: String, optional: true },
+    };
+
+    setup() {
+        this.state = useState({
+            barcode: false,
+        });
+        this.barcodeManual = useRef('manualBarcode');
+        // Autofocus processing was blocked because a document already has a focused element.
+        onMounted(() => {
+            this.barcodeManual.el.focus();
+        });
```

Comment:
[useAutofocus](https://github.com/odoo/odoo/blob/2f0b96bf1ccf89ee39f9fddb0ef6b18791700f9d/addons/web/static/src/core/utils/hooks.js#L38) is your friend here :wink:

Reply:
useAutofocus is not working here, maybe some other document is already focused.

Reply:
Hello @FlorianGilbert any insight on this ? 
also for the below comment https://github.com/odoo/odoo/pull/171934#issuecomment-2355160437